clusterloader2/api/types.go

```diff
@@ -54,20 +54,41 @@ type Config struct {
 	ChaosMonkey ChaosMonkeyConfig `json:"chaosMonkey"`
 }
 
-// Step represents encapsulation of some actions. These actions could be
-// object declarations or measurement usages.
-// Exactly one field (Phases or Measurements) should be non-empty.
+// Step represents a unit of work in ClusterLoader2. It can be either:
+// - a collection of measurements,
+// - a collection of phases,
+// - a group of other steps (module).
+// Exactly one field (Phases or Measurements or Module) should be non-empty.
```

Comment:
Correct me If I'm wrong, but I don't see a place where this behaviour is validated (Making sure that only one of 3 is non-empty).  What I propose is replacing IsMeasurement/IsPhase/IsModule with enum + method GetType that also validates if this requirement is fulfilled (or some similar solution). What do you think about it?

Reply:
We don't validate it directly rather check during execution. Here - https://github.com/kubernetes/perf-tests/blob/f0c628601e7721063a28dda50df422a90cabb446/clusterloader2/pkg/test/simple_test_executor.go#L169 and in the code that flattens module steps where we process all "module" steps.

I think it'd be better if we validated the config before we start executing it. It will result in a cleaner code and better user experience (test will fail fast, not in the middle of Xh run). But it's a separate issue. 

Introducing an enum would be a backward incompatible change, so probably not a good way to go.

Reply:
@marseel, would you mind opening an issue for this (cl2 config validation) ?

Reply:
Sure, I've opened it. It's exactly what I had in mind - cleaner code and better user experience.

Reply:
Thanks!